### ISSUE TITLE
Solution configuration is invalid

### ISSUE THREAD

**Issue by TrianglesPCT** [slynch8/10x#326]
I tried opening my existing VS solution and building, but I get this error(doesn't happen using Visual Studio).

**Here is the command it used when I tried to build just the primary DLL(makemake).**
msbuild /t:"6_ Game\makemake" /p:Configuration="Deploy Win64_AVX2" /p:Platform="x64" /verbosity:minimal "C:/depo/common/build_premake17/Cloud.sln" /m

**And the error**
C:\depo\common\build_premake17\Cloud.sln.metaproj : error MSB4126: The specified solution configuration "Deploy Win64_AVX2|x64" is invalid. Please specify a valid solution configuration using the Configuration and Platform properties (e.g. MSBuild.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") or leave those properties blank to use the default solution configuration. [C:\depo\common\build_premake17\Cloud.sln]

BTW I like the editor, it is much more responsive than VS:) Multi-monitor would be super great, kinda crammed with everything in one view:(

**Comment by slynch8** [comment]
It's difficult to know why it is failing without seeing the project. Could you share the project or re-create the issue in a simple test project?

Another thing to try is see if you can get it working from the command line. Open a VS2019 prompt and paste the msbuild command line above. Try changing the Configuration option. Sometimes solution configurations don't match up with project configurations although I thought I had fixed this.

Thanks, glad you like 10x. Yes, multi-monitor is planned.

**Comment by TrianglesPCT** [comment]
I believe one thing that is wrong is setting platform to x64, if I change the command to:

![image](https://user-images.githubusercontent.com/1130993/128118437-deac8b0c-6884-4f5a-9f74-d60b07efb929.png)

msbuild /t:"6_ Game\makemake" /p:Configuration="Deploy" /p:Platform="Win64_AVX2" /verbosity:minimal "C:/depo/common/build_premake17/Cloud.sln" /m

It fails on something else: 
 error MSB4057: The target "6_ Game\makemake" does not exist in the project.

**Comment by slynch8** [comment]
It looks like the config and platform are correct in your screengrab of the drop down boxes. Does it work if you do "Build All" instead of "Build Active Workspace". Build All will build the solution. I'm wondering if the project is set to the wrong configuration. In the .sln file there is a section that looks like this:

```
GlobalSection(ProjectConfigurationPlatforms) = postSolution
        {BEA64C37-7397-4301-8885-6A2DF436F5F9}.Debug_notepad|Any CPU.ActiveCfg = Debug_notepad|x64
        {BEA64C37-7397-4301-8885-6A2DF436F5F9}.Debug_notepad|x64.ActiveCfg = Debug_notepad|x64
        {BEA64C37-7397-4301-8885-6A2DF436F5F9}.Debug_notepad|x64.Build.0 = Debug_notepad|x64
        {BEA64C37-7397-4301-8885-6A2DF436F5F9}.Debug_notepad|x86.ActiveCfg = Debug_notepad|x64
        etc.
```

It declares the config/platform for each project for each different solution config/platform. See what the project is mapped to for that config/platform (you'll have to lookup the GUIDs further up the file).

**Comment by TrianglesPCT** [comment]
Here are is the section for the makemake project

		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_AVX.ActiveCfg = Deploy Win64_AVX|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_AVX.Build.0 = Deploy Win64_AVX|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_AVX2.ActiveCfg = Deploy Win64_AVX2|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_AVX2.Build.0 = Deploy Win64_AVX2|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_AVX512.ActiveCfg = Deploy Win64_AVX512|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_AVX512.Build.0 = Deploy Win64_AVX512|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_SSE2.ActiveCfg = Deploy Win64_SSE2|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_SSE2.Build.0 = Deploy Win64_SSE2|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_SSE4.ActiveCfg = Deploy Win64_SSE4|x64
		{4109853F-2DD6-DF4A-56EF-CDF2421B15D9}.Deploy|Win64_SSE4.Build.0 = Deploy Win64_SSE4|x64

If I build the entire solution it does work, only fails when trying to build projects individually(I tried other projects, they all seem to fail in the same way as makemake).

**Comment by slynch8** [comment]
I assume it would fail in VS as well if you try and build a single project? The project is setup to use x64 for the platform so it looks like 10x is doing the right thing.

What does your Configuration manager look like in VS?

**Comment by TrianglesPCT** [comment]
Building individual projects works in VS.

The screenshot for the configuration is from VS actually, it looks the same in 10x.

I made a copy of the solution + makemake project, it doesn't have any source code, or any of the other projects, but it is enough to trigger the error. 

1. Unzip
2. go to error/build and open Cloud.sln with 10x
3. right click on Game->makemake and hit build.

[error.zip](https://github.com/slynch8/10x/files/6932236/error.zip)

**Comment by slynch8** [comment]
Thanks for that, yes, I'm getting it with that project.

I just noticed, it looks like the project name is wrong "6_ Game\makemake", it doesn't like the . in the project name. I'll get that fixed.

**Comment by slynch8** [comment]
ah! think I've got it. Spaces need to be escaped with %20

Can you try this:
msbuild /t:"6_%20Game\makemake" /p:Configuration="Deploy Win64_AVX" /p:Platform="x64" /verbosity:minimal "D:/temp/error/error/build/Cloud.sln" /m

**Comment by TrianglesPCT** [comment]
That seems to have at least gotten MSBuild to try and run it, but when I enter that command into the Powershell command inside VS this is what I get:

 msbuild /t:"6_%20Game\makemake" /p:Configuration="Deploy Win64_AVX" /p:Platform="x64" /verbosity:minimal "C:/depo/common/build_premake17/Cloud.sln" /m
Microsoft (R) Build Engine version 16.10.0+4242f381a for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Build.Construction.SolutionProjectGenerator.CreateTraversalInstance(String wrapperProjectToolsVersion, Boolean explicitToolsVersionSpecified, List`1 projectsInOrder)
   at Microsoft.Build.Construction.SolutionProjectGenerator.CreateSolutionProject(String wrapperProjectToolsVersion, Boolean explicitToolsVersionSpecified)
   at Microsoft.Build.Construction.SolutionProjectGenerator.Generate()
   at Microsoft.Build.Execution.ProjectInstance.GenerateSolutionWrapper(String projectFile, IDictionary`2 globalProperties, String toolsVersion, ILoggingService loggingService, BuildEventContext projectBuildEventContext, IReadOnlyCollection`1 targetNames, ISdkResolverService sdkResolverService, Int32 submissionId)
   at Microsoft.Build.Execution.ProjectInstance.LoadSolutionForBuild(String projectFile, PropertyDictionary`1 globalPropertiesInstances, String toolsVersion, BuildParameters buildParameters, ILoggingService loggingService, BuildEventContext projectBuildEventContext, Boolean isExplicitlyLoaded, IReadOnlyCollection`1 targetNames, ISdkResolverService sdkResolverService, Int32 submissionId)
   at Microsoft.Build.Execution.BuildManager.LoadSolutionIntoConfiguration(BuildRequestConfiguration config, BuildRequest request)
   at Microsoft.Build.Execution.BuildManager.HandleNewRequest(Int32 node, BuildRequestBlocker blocker)
   at Microsoft.Build.Execution.BuildManager.IssueBuildSubmissionToScheduler(BuildSubmission submission, Boolean allowMainThreadBuild)
   at Microsoft.Build.Execution.BuildManager.<>c__DisplayClass79_0.<ExecuteSubmission>b__7()

MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Boolean needToVali   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Boolean needToValidateProject, String schemaFile, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, TextWriter targetsWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, GraphBuildOptions graphBuildOptions, Boolean lowPriority, String[] inputResultsCaches, String outputResultsCache)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String commandLine)

Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Boolean needToValidateProject, String schemaFile, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, TextWriter targetsWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, GraphBuildOptions graphBuildOptions, Boolean lowPriority, String[] inputResultsCaches, String outputResultsCache)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String commandLine)
   at Microsoft.Build.CommandLine.MSBuildApp.Main()

**Comment by slynch8** [comment]
It might be that the Powershell in VS is already escaping, not sure. Does it work from a VS2019 cmd prompt?

**Comment by TrianglesPCT** [comment]
I tried the cmd prompt also now, but it spits out the same thing

**Comment by slynch8** [comment]
I'm not sure what else to suggest. An "internal error" in msbuild doesn't bode well. Can you get by for now using Build All?

**Comment by TrianglesPCT** [comment]
I'll take another stab at the MSBuild docs another day, I can get by with the solution build

**Comment by slynch8** [comment]
Ok. Might be worth reporting it here: https://github.com/dotnet/msbuild/issues

**Comment by slynch8** [comment]
added %20 fix in (0.21.21)

**Comment by slynch8** [comment]
I've removed the %20 fix as this wasn't the right fix and breaks it for normal projects with spaces. There is something special about this sln, I'm guessing it's because the project is in a folder with a special '.' char and/or a space.

In summary, msbuild seems to ignore the project name and tries to build the sln, but the sln doesn't have the specified config (the project does). Removing or replacing the space makes msbuild product an internal error but I think that is a separate issue.

Closing this now as I can't think of anything else to try. Please rename your projects or report it to https://github.com/dotnet/msbuild/issues
